articles/understanding-map-reduce-in-hadoop/index.md

```diff
@@ -0,0 +1,118 @@
+- **Input data:** This is the data used for processing in the mapping phase.
+- **Output data:** This is the result of mapping and reducing.
+- **Client:** The MapReduce receives jobs from clients. It can accept jobs from many clients. 
+- **Hadoop MapReduce Master:** This plays the role of dividing jobs into job-parts.
+- **Job-parts:** These are sub-jobs that result from the division of the main job. 
+
+In the MapReduce architecture, jobs are submitted by clients to the MapReduce Master. This master will then sub-divide the job into equal sub-parts. The job-parts will be used for the two main tasks in MapReduce: mapping and reducing. The developer will write logics that satisfies the requirements of the organization or company. The input data will be split and mapped.
+
+The intermediate data will then be sorted and merged. The resulting output will be processed by the reducer, which will generate a final output stored in the HDFS.
+
+The following diagram shows a simplified flow diagram for the MapReduce program. 
+
+![MapReduce Flow Diagram](/engineering-education/understanding-mapreduce-in-hadoop/mapreduce-flow-diagram.jpg)
+
+[Image Source: Data Flair](https://data-flair.training/blogs/wp-content/uploads/sites/2/2017/09/hadoop-mapreduce-flow.jpg)
+
+#### How Job Trackers and Task Trackers Work
+Every job consists of two main components: mapping task and reducing task. The map task plays the role of splitting jobs into job-parts and mapping intermediate data. The reduce task plays the role of shuffling and reducing intermediate data into smaller units. 
+
+The job tracker acts as a master. It ensures that all jobs are executed. The job tracker schedules jobs that have been submitted by clients. It will assign jobs to task trackers. Each task tracker consists of a map task and reduces the task. Task trackers report the status of each assigned job to the job tracker. The following diagram provides a summary of how job trackers and task trackers work. 
+
+![Job Trackers and Task Trackers](/engineering-education/understanding-mapreduce-in-hadoop/job-trackers-and-task-trackers.png)
+
+[Image Source: CNBlogs](https://images2015.cnblogs.com/blog/222972/201512/222972-20151215101500256-2116703561.png)
+
+#### Phases of MapReduce
+The MapReduce program is executed in three main phases: mapping, shuffling, and reducing. There is also an optional phase known as the combiner phase.
+
+##### Mapping Phase
+This is the first phase of the program. There are two steps in this phase: splitting and mapping. In this phase, a dataset is split into equal units called chunks. These chunks can be termed as input splits. MapReduce uses the input splits as inputs. They are composed of two parts: keys and values. In the mapping step, the required action is performed using these subsets. This step contains a coding logic that is applied to a specific number of data blocks. The output of this phase is in the form of keys and values (k, v). 

```

Comment:
What are you referring to when you say they are composed of two parts: keys and values? What is 'they'? Also, after they are composed of two parts, this para is a bit difficult to comprehend. Can you use the previous example that I have outlined and make the changes accordingly?

Reply:
I have revised this section.